lib/rubocop/cop/naming/constant_wording.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Naming
+      # This cop checks whether constant names are written using
+      # clearer wording.
+      #
+      # @example
+      #   # bad
+      #   Whitelist = []
+      #   WhiteList = []
+      #   Whitelisted = []
+      #   Blacklist = []
+      #   BLACKLISTED = []
+      #
+      #   # good
+      #   Allowlist = []
+      #   PermittedList = []
+      #   Denylist = []
+      class ConstantWording < Cop
+        MSG = 'Please use clearer names for constants.'.freeze
+
+        def on_casgn(node)
+          constant_name = get_constant_name(node)
+
+          add_offense(node, location: :name) if not_clear?(constant_name)
+        end
+
+        private
+
+        NAME_DENY_LIST = %i[
```

Comment:
I think you should make the denylist a parameter, so people can extend this to other names themselves. We can actually make this a map, so we can suggest good name instead of the problematic names.

Reply:
Sounds good to me :) Any existing example of passing in a parameter map?

Reply:
`Style/CollectionMethods`.